### PR TITLE
Prevent the creation of new branches from random users

### PR DESCRIPTION
**Pull request by Girgias**
Because I've done this too many times

### CODE REVIEW

**Comment by bwoebi** [comment]
There has been normal usage of non-release branches on php-src - maybe we can prefix them with something like `dev/somebranch` - then you definitely don't do it un-intentionally.

File: hooks/pre-receive [hunk 1]
```diff
@@ -156,6 +156,14 @@ if (empty($req_paths)) {
     $req_paths = ['']; // we can empty paths for trivial merge
 }
 
+if ($pi->isNewBranch()) {
+    $authorizedUsers = $RMs + $weKnowWhatWeAreDoing;
```
Comment (nikic) [inline hooks/pre-receive:160]: This doesn't do what you want it to do...

**Comment by nikic** [comment]
This should probably be limited to php-src only?

**Comment by nikic** [comment]
> There has been normal usage of non-release branches on php-src - maybe we can prefix them with something like `dev/somebranch` - then you definitely don't do it un-intentionally.

I don't think we've used feature branches on php-src in a very long time. Nowadays it's expected that they will be on a fork.

**Pull request closed without merging**